### ISSUE TITLE
plugin load fails to parse ReportRelative config option

### ISSUE THREAD

**Issue by faxm0dem** [collectd/collectd#3180]
When using the option `ReportRelative` in the load plugin on 5.9.0, the config parsing fails, and collectd refuses to start:

```
> cat /tmp/collectd.conf
LoadPlugin load
<Plugin "load">
  ReportRelative true
</Plugin>
> collectd -C /tmp/collectd.conf -T
plugin_load: plugin "load" successfully loaded.
Error: Parsing the config file failed!
```

**Comment by faxm0dem** [comment]
here's the debug output:

```
☠ ./collectd -C load.conf -T
option = ReadThreads; value = -1;
Created new plugin context.
plugin_load: plugin "load" successfully loaded.
[2019-06-18 16:44:45] plugin_load: plugin "logfile" successfully loaded.
[2019-06-18 16:44:45] type = logfile, key = LogLevel, value = debug
[2019-06-18 16:44:45] type = logfile, key = File, value = STDOUT
[2019-06-18 16:44:45] type = load, key = ReportRelative, value = true
[2019-06-18 16:44:45] Done parsing `/opt/collectd/share/collectd/types.db'
Error: Parsing the config file failed!
```

**Comment by faxm0dem** [comment]
git-bisect++ this was introduced by 3b9c7b21b4ddfcf59a0147bc9e91e8889ca78d56

**Comment by faxm0dem** [comment]
turns out other plugins' `load_config` return non-zero status.
The change mentioned earlier was a bit dangerous - I'll try to find other corner cases tomorrow, or should we simply revert 3b9c7b2?

**Comment by rpv-tomsk** [comment]
>The change mentioned earlier was a bit dangerous - I'll try to find other corner cases tomorrow, or should we simply revert 3b9c7b2?

That is not dangerous change and we should not revert 3b9c7b2.

**Comment by rpv-tomsk** [comment]
As a maintainer without code writing privilege, I can't / willn't write PR to fix this issue.
So, anybody should do this and I should wait for PR.

#2958

https://github.com/orgs/collectd/teams/trusted-contributors/discussions/1

**Comment by faxm0dem** [comment]
I'm currently looking at all config callbacks, and trying to see if their RC is correct.
As of now I found `load` and `logfile` that misbehave.

I'll post questions here as I go.

Here's the first one: in barometer plugin, we have this:

```
  } else if (strcasecmp(key, "Normalization") == 0) {
    int normalize_tmp = atoi(value);
    if (normalize_tmp < 0 || normalize_tmp > 2) {
      WARNING("barometer: collectd_barometer_config: invalid normalization: %d",
              normalize_tmp);
      return 1;
    }
```

WARNING and rc=1 are inconsistent, should we either:

* s/WARNING/ERROR/
or
* s/RC=1/RC=0/

?

**Comment by faxm0dem** [comment]
it seems to me most plugins return negative values (mostly -1) on failure and positive on warnings

**Comment by faxm0dem** [comment]
What I chose to do:

* fix RC when I judged it to be non-fatal (load, logfile)
* fix loglevel from WARNING to ERROR in other cases (barometer, openvpn, ted)
* add log message (syslog)

**Comment by rpv-tomsk** [comment]
Collectd has `-t` option which intended to do config test.
At some point developers decided to handle config errors in a 'soft' way, which makes this option useless.

Maybe we had to reimplement config test using positive and negative return codes?

**Comment by faxm0dem** [comment]
the `-T` option does as intended, and I used it for `git bisect`

**Comment by faxm0dem** [comment]
also, from going through the code of all plugins, I can say that most use `-1` when the option isn't supported, and a positive value for all other failed cases, e.g. invalid value for option.
If we want a policy regarding this, maybe that's the guideline to push?

**Comment by faxm0dem** [comment]
on a sidenote, this fix should definitely trigger a 5.9.1 release as load plugin is 50% broken
